browser/brave_wallet/keyring_controller_unittest.cc

```diff
@@ -1741,4 +1744,58 @@ TEST_F(KeyringControllerUnitTest, AddAccountsWithDefaultName) {
   run_loop.Run();
 }
 
+TEST_F(KeyringControllerUnitTest, SignMessageByDefaultKeyring) {
+  // HDKeyringUnitTest.SignMessage already tests the correctness of signature
+  KeyringController controller(GetPrefs());
+  controller.RestoreWallet(kMnemonic1, "brave", false,
+                           base::DoNothing::Once<bool>());
+  base::RunLoop().RunUntilIdle();
+  ASSERT_FALSE(controller.IsLocked());
+
+  std::string account1;
+  {
+    base::RunLoop run_loop;
+    controller.GetDefaultKeyringInfo(
+        base::BindLambdaForTesting([&](mojom::KeyringInfoPtr keyring_info) {
+          ASSERT_EQ(keyring_info->account_infos.size(), 1u);
+          account1 = keyring_info->account_infos[0]->address;
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+  const std::vector<uint8_t> message = {0xde, 0xad, 0xbe, 0xef};
+  auto sig_with_err = controller.SignMessageByDefaultKeyring(account1, message);
+  EXPECT_NE(sig_with_err.signature, absl::nullopt);
+  EXPECT_FALSE(sig_with_err.signature->empty());
+  EXPECT_TRUE(sig_with_err.error_message.empty());
+
+  // message is 0x
+  sig_with_err =
+      controller.SignMessageByDefaultKeyring(account1, std::vector<uint8_t>());
+  EXPECT_NE(sig_with_err.signature, absl::nullopt);
+  EXPECT_FALSE(sig_with_err.signature->empty());
+  EXPECT_TRUE(sig_with_err.error_message.empty());
+
+  // not a valid account in this wallet
+  const std::vector<std::string> invalid_accounts(
+      {"0xea3C17c81E3baC3472d163b2c8b12ddDAa027874", "", "0x1234"});
+  for (const auto& invalid_account : invalid_accounts) {
+    sig_with_err =
+        controller.SignMessageByDefaultKeyring(invalid_account, message);
+    EXPECT_EQ(sig_with_err.signature, absl::nullopt);
+    EXPECT_EQ(
+        sig_with_err.error_message,
+        l10n_util::GetStringFUTF8(IDS_BRAVE_WALLET_SIGN_MESSAGE_INVALID_ADDRESS,
+                                  base::ASCIIToUTF16(invalid_account)));
```

Comment:
Will this fail for people with different locale settings? If so maybe just leave out the error message check

Reply:
no, it won't fail